From 09a4253ff82f71d5de711b561b8c4b9f73b48f50 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 9 May 2015 14:49:19 +0100
Subject: [PATCH 2/2] Debugger: Handle case of os.path.sep being '/'

On MSYS2, if QtCreator is launched from one of the msys2 shells
then the MSYSTEM env. var is set and our Pythons set os.sep and
os.path.sep to '/' so that it aligns with the default separator
of those shells.

fromNativePath(str) then fails to convert '\' seprators embedded
in binaries since str.replace('/', '/') is a noop. The operation
we want here is to convert all '\' to '/' anyway so this is more
explicit regardless of any opinions on our choice to dynamically
vary os.sep and os.path.sep

Change-Id: I76c05b188abca41c6a1b516c68b584e0c06e7235
---
 share/qtcreator/debugger/gdbbridge.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/share/qtcreator/debugger/gdbbridge.py b/share/qtcreator/debugger/gdbbridge.py
index d219689..e96fe86 100644
--- a/share/qtcreator/debugger/gdbbridge.py
+++ b/share/qtcreator/debugger/gdbbridge.py
@@ -1545,7 +1545,7 @@ class Dumper(DumperBase):
 
     def stackListFrames(self, args):
         def fromNativePath(str):
-            return str.replace(os.path.sep, '/')
+            return str.replace('\\', '/')
 
         limit = int(args['limit'])
         if limit <= 0:
-- 
2.4.0

